rfcs/0024-nondeterministic-functions.md

```diff
@@ -0,0 +1,60 @@
+---
+feature: non_deterministic_functions
+authors:
+  - "st1page"
+start_date: "2022/11/21"
+---
+
+# nondeterministic-functions
+
+## Summary
+
+We introduce the `MaterializeProject` stream executor to handle non-deterministic function on updatable stream.
+
+## Motivation
+
+In the discussion about the udf, @skyzh raise the issue about the non-deterministic function on updatable stream <https://github.com/risingwavelabs/rfcs/pull/12#issuecomment-1305105614>. This issue also appears on the built-in functions such as `random(),` `proc_time()` or `now()`.
```

Comment:
For UDF, how can we decide the determinism? Specified by users? 🤔

Reply:
BTW, from @jon-chuang's demo idea, is "caching expensive UDF" another valid motivation? 🤔

Reply:
Can we trust it? We currently maintain very high consistency in our system and a wrong old value could produce unsolvable panic... We even do not totally trust the CDC source now... :thinking:

Reply:
Then always materializing sounds good to me 🤪

Reply:
Caching is an application-level concept... the user needs to be sure they are OK with stale values.

So yes, updateable stream is root cause.

Reply:
Agreed on always materializing